src/Lib/App/Error.hs

```diff
@@ -9,22 +9,28 @@ module Lib.App.Error
 
 import Control.Monad.Except (MonadError, throwError)
 
-data AppError =
-    Invalid Text
-  | NotAllowed Text
-  | NotFound
-  | ServerError Text
-  deriving (Show, Eq)
+data AppError = InternalError IError deriving (Show, Eq)
 
 -- | Type alias for errors.
 type WithError m = MonadError AppError m
 
--- Extract the value from a maybe, throwing the given 'AppError' if
--- the value does not exist
+-- | Extract the value from a maybe, throwing the given 'AppError' if
+-- | the value does not exist
```

Comment:
You don't need to put `|` before each line. Only after first `--`. Like here:

* https://github.com/Holmusk/CDMP/blob/cd24a6fb3f3d187f68bdd0d8ff7056ba456f5f58/backend/src/CDMP/Db.hs#L27-L30